src/api/customer/customer-type.ts

```diff
@@ -45,4 +45,67 @@ export type CustomerCreationRequestHandler = RequestHandler<
         customerCreated: CustomerAttributes,
         customerToRespond: CustomerResponse
     }
->;
\ No newline at end of file
+>
+
+type CustomerFindError = {
+    statusCode: string;
+    message: string;
+}
+
+type CustomersFindByPk = {
+    uuid: string;
+}
+
+type CustomersFindRequest = {
+    name: string;
+    document: {
+        cpf?: string;
+        cnpj?: string;
+    },
+    contact: {
+        email: string;
+        phone?: string;
+    }
+} 
+
+type CustomersFindResponse = CustomersFindRequest & {
+    uuid: string;
+    createdAt: string;
+    updatedAt: string;
+}
+
+export type CustomersFindRequestHandler = RequestHandler<
+    CustomersFindByPk, // path params
+    CustomersFindResponse | CustomerFindError,  // response
+    CustomersFindRequest, // request
+    {}, // query params
+    {
+        customerToPK: CustomersFindByPk
+        customerFindError: CustomerFindError,
+        customerFind: CustomerAttributes,
+        customerToRespond: CustomersFindResponse
+    }
+>
+
+
+type CustomerFindAll = Array<CustomerAttributes>;
```

Comment:
Legal, só por curiosidade, no typescript essa outra anotação é mais comum:

```suggestion
type CustomerFindAll = CustomerAttributes[];
```